### PR TITLE
Improve VS retry documentation

### PR DESCRIPTION
**Pull request by howardjohn**
Inspired by https://github.com/istio/istio/issues/51762

### CODE REVIEW

**Comment by Stono** [comment]
Lgtm!

**Review by zirain — APPROVED** [review]

File: networking/v1alpha3/virtual_service.proto [hunk 1]
```diff
@@ -1272,6 +1272,14 @@ message StringMatch {
 //       retryOn: gateway-error,connect-failure,refused-stream
 // ```
 //
+// Note: the default cluster-wide retry policy, if not specified, is:
+//
+// ```yaml
+// attempts: 2
```
Comment (hzxuzhonghu) [inline networking/v1alpha3/virtual_service.proto:1278]: move this below to the `retry_on`  comments, if  this struct shared, it could not be right anymore

Comment (howardjohn) [reply networking/v1alpha3/virtual_service.proto:1278]: It's not the default for retry_on though but for the whole thing. Maybe onto `retries`?

Comment (hzxuzhonghu) [reply networking/v1alpha3/virtual_service.proto:1278]: right, that's ok

Comment (howardjohn) [reply networking/v1alpha3/virtual_service.proto]: ok moved it over

**Review by hzxuzhonghu — APPROVED** [review]

**Review by ramaraochavali — APPROVED** [review]

**Comment by craigbox** [comment]
cc @linsun @therealmitchconnors for TOC approval

**Review by therealmitchconnors — APPROVED** [review]

**Pull request merged**